tsdb/cursors/schema.go

```diff
@@ -1,20 +1,89 @@
 package cursors
 
+import "github.com/influxdata/influxql"
+
 // FieldType represents the primitive field data types available in tsm.
 type FieldType int
 
 const (
 	Float     FieldType = iota // means the data type is a float
 	Integer                    // means the data type is an integer
 	Unsigned                   // means the data type is an unsigned integer
-	Boolean                    // means the data type is a boolean
 	String                     // means the data type is a string of text
+	Boolean                    // means the data type is a boolean
 	Undefined                  // means the data type in unknown or undefined
 )
 
+var (
+	fieldTypeToDataTypeMapping = [8]influxql.DataType{
+		Float:     influxql.Float,
+		Integer:   influxql.Integer,
+		Unsigned:  influxql.Unsigned,
+		String:    influxql.String,
+		Boolean:   influxql.Boolean,
+		Undefined: influxql.Unknown,
+		6:         influxql.Unknown,
+		7:         influxql.Unknown,
+	}
+)
+
+// IsLower returns true if the other FieldType has greater precedence than the
+// current value. Undefined has the lowest precedence.
+func (ft FieldType) IsLower(other FieldType) bool { return other < ft }
+
+// FieldTypeToDataType returns the equivalent influxql DataType for the field type ft
+func FieldTypeToDataType(ft FieldType) influxql.DataType {
+	return fieldTypeToDataTypeMapping[ft&7]
```

Comment:
@jacobmarble regarding this comment:

> They are fillers in the 8-element array to ensure the if an unspecified value is indexed in the function, it always returns `influxql.Unknown`

This is the code I was referring to. If an invalid `FieldType` is specified, the results are undefined, but the 7th and 8th elements (index 6, 7) are explicitly set to `influxql.Unknown`